### ISSUE TITLE
L046 show up as a fixable rule but isn't fixable

### ISSUE THREAD

**Issue by cooley-hi** [sqlfluff/sqlfluff#4311]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When running `sqlfluff fix` L046 shows up as a fixable rule, but per the docs it is not fixable and does not actually get fixed -- at least not with dbt. It does appear to work for non-ref tags, but doesn't work for the ref tags.

### Expected Behaviour

When given something like
```sql
select *
from {{ref('some table')}}
```

you get

```sql
select *
from {{ ref('some_table') }}
```

OR it doesn't list it as a fixable error. Either way the docs should be updated.

### Observed Behaviour

When given something like
```sql
select *
from {{ref('some table')}}
```

Sqfluff will list the second line as a violation and offer to fix it, but it won't fix it.

### How to reproduce

1. Run sqlfluff fix on a model with a `{{ref(...)}}`
2. Confirm you want to fix

### Dialect

snowflake

### Version

sqlfluff==1.4.5
sqlfluff-templater-dbt==1.4.5

### Configuration

`.sqlfluff`

```
[sqlfluff]
dialect = snowflake
templater = dbt
rules = L001, L003, L004, L005, L006, L008, L009, L010, L011, L012, L013, L014, L017, L018, L019, L023, L024, L027, L030, L040, L043, L045, L046, L060, L061, L063
large_file_skip_byte_limit = 100000

[sqlfluff:rules]
tab_space_size = 2
max_line_length = 100
indent_unit = space

[sqlfluff:layout:type:comma]
line_position = leading

[sqlfluff:indentation]
indented_joins = true

[sqlfluff:templater:jinja]
apply_dbt_builtins = true

[sqlfluff:rules:L010]
capitalisation_policy = lower

[sqlfluff:rules:L014]
extended_capitalisation_policy = lower

[sqlfluff:rules:L030]
extended_capitalisation_policy = lower

[sqlfluff:rules:L040]
capitalisation_policy = lower

[sqlfluff:rules:L063]
extended_capitalisation_policy = lower
```

`.sqlfluffignore`

```
target/
dbt_packages/
macros/
snapshots/
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by pdebelak** [comment]
For what it's worth, this at least appears to work with the jinja templater:

```shell
$ cat .sqlfluff 
[sqlfluff]
dialect = snowflake
templater = jinja
$ echo "select *
from {{ref('some_table')}}" | sqlfluff fix -
select *                                                                                                                                                               
from {{ ref('some_table') }}
```

**Comment by peterbud** [comment]
With the latest sqlfluff, this rule issue is now categorized as fixable:
```bash
sqlfluff fix test.sql -d postgres 
==== finding fixable violations ====
== [test.sql] FAIL
L:   2 | P:   6 | JJ01 | Jinja tags should have a single whitespace on either
                       | side: {{ref('some_table')}} [jinja.padding]
L:   2 | P:  27 | LT12 | Files must end with a single trailing newline.
                       | [layout.end_of_file]
== [test.sql] FIXED
2 fixable linting violations found
```
